File: Sources/Server/KioskBrains.Server.Web/src/models/base/base-search-request.ts
import { SearchRequestMetadata } from '../base/search-request-metadata';

export class BaseSearchRequest<T extends object> {
    constructor() {
        this.metadata = new SearchRequestMetadata();
        this.searchStruct = {} as T;
    }

    public searchTerm: string;
    public searchStruct: T;
    public isAdvancedSearch: boolean;
    public metadata: SearchRequestMetadata;
}

File: Sources/Server/KioskBrains.Server.Domain/Actions/Common/Models/BaseSearchRequest.cs
namespace KioskBrains.Server.Domain.Actions.Common.Models
{
    public class BaseSearchRequest<TForm>
    {
        public string SearchTerm { get; set; }

        public TForm SearchStruct { get; set; }

        public bool IsAdvancedSearch { get; set; }

        public SearchMetadata Metadata { get; set; }
    }
}